core/controllers/feedback.py

```diff
@@ -282,3 +282,29 @@ def post(self):
         email_manager.send_suggestion_email(
             exploration.title, exploration.id, suggestion.author_id,
             exploration_rights.owner_ids)
+
+
+class InstantFeedbackEmailHandler(base.BaseHandler):
+    """Handles task of sending feedback message emails instantsly."""
+
+    def post(self):
+        payload = json.loads(self.request.body)
+        user_id = payload['user_id']
+        reference = payload['reference']
+        message = feedback_services.get_message(
+            reference['exploration_id'], reference['thread_id'],
+            reference['message_id'])
+        exploration = exp_services.get_exploration_by_id(
+            reference['exploration_id'])
+        thread = feedback_services.get_thread(
+            reference['exploration_id'], reference['thread_id'])
+
+        if 'updated_status' in payload:
```

Comment:
Okay. Let me explain what happens here. Suppose that someone sent feedback and also committed a change in same feedback message. Now, if look at create_message() code, first it will check that whether status is updated or not. In this case yes it is changed and therefore `add_feedback_thread_status_change_email` function will be called. This function will enqueue as task that has `'updated_status' in payload`. So task will execute calling the handler and ultimately and email will be sent notifying that `author changed subject to X from Y`.

While this is happening in parallel thread our create_message is still going on. Now it comes to regular feedback message check and sees that message length is not 0. Thus it will call `add_message_to_email_buffer` function. This function now will enqueue a task for sending email that has nothing to do with status change and payload generated for this task does not contain `'updated_status' in payload`. Thus the email will be sent containing feedback message from author. 

So all in all, first we check that status has changed or not and if it is then enqueue the task for sending email about this change. This task has nothing to do with message text because if status is changed then (if you see how handler is designed) text which is sent in email is about change in status. And then we do regular check whether message is empty or not and if it is not empty then send email for this feedback message. At this time we have nothing to do with message text because `'updated_status' is not in payload`. I hope it makes clear to understand how it is actually going on. I will write one test case in feedback_test for it too (I forgot this case).

But looking at the code gives a feeling that it will not send feedback message when status is also changed. If you think that code should be designed in some other manner then suggestions are welcome.

Thanks